src/main/java/org/carlmontrobotics/robotcode2023/commands/RunRoller.java

```diff
@@ -12,31 +11,30 @@
 public class RunRoller extends CommandBase {
 
     private final Roller roller;
-    private final Color ledColor;
     private final Timer timer = new Timer();
     private final RollerMode mode;
 
-    public RunRoller(Roller roller, RollerMode mode, Color ledColor) {
+    public RunRoller(Roller roller, RollerMode mode) {
         addRequirements(this.roller = roller);
         this.mode = mode;
-        this.ledColor = ledColor;
     }
 
     @Override
     public void initialize() {
-        roller.setSpeed(mode.speed);
-        roller.setLedColor(ledColor);
+        if(roller.hasGamePiece() && isIntake()) cancel();
+        if(!roller.hasGamePiece() && !isIntake() && mode.speed > 0) cancel(); // is not stopping the rollers then cancel
```

Comment:
Just do `if(mode.speed > 0)` around the whole block in case someone passes intake true speed = 0; (For some reason)